pandas/_testing/asserters.py

```diff
@@ -1344,6 +1344,7 @@ def assert_frame_equal(
                 rtol=rtol,
                 atol=atol,
                 check_index=False,
+                check_flags=check_flags,
```

Comment:
right. I think I am starting to get my head round this. This should be `check_flags=False` to override the checking in assert_series_equal and be consistent with `check_index=False` in the line above.

the flags check *is* already performed on the DataFrame.

Reply:
Yep you are correct. Passing in False is the way to go here